### PR TITLE
Expanded github actions testing

### PR DESCRIPTION
**Pull request by brtnfld**
I added the additional tests for different versions of zfp:
(1) test with the develop version
(2) test with the latest tagged release (is 1.0.0)
(3) test with pre-1.0.0 release

I also changed the Autotools tests to not issue a fail for skipped tests.
Changed to 1.12.2. for windows testing.

### CODE REVIEW

File: src/H5Zzfp.c [hunk 1]
```diff
@@ -29,16 +29,16 @@ and calls to bitstream methods with 'B ' as in
 #undef B
 #endif
 
+#include "zfp.h"
```
Comment (markcmiller86) [inline src/H5Zzfp.c:32]: We need to keep this as it was because it effects how ZFP and H5Z-ZFP get built into Silo

**Comment by brtnfld** [comment]
The current tests will fail with zfp 0.5.3, https://github.com/brtnfld/H5Z-ZFP/runs/7908035339?check_suite_focus=true

```
gcc -c test_error.c -o test_error.o -fPIC -DZFP_LIB_VERSION=0x053 -I../src -I/home/runner/work/H5Z-ZFP/H5Z-ZFP/zfp/build/zfp/include -I/usr/include/hdf5/serial
gcc test_error.o -o test_error -fPIC -Wl,-rpath,/usr/lib/x86_64-linux-gnu/hdf5/serial -Wl,-rpath,/home/runner/work/H5Z-ZFP/H5Z-ZFP/zfp/build/zfp/lib -L../src -L/usr/lib/x86_64-linux-gnu/hdf5/serial -L/home/runner/work/H5Z-ZFP/H5Z-ZFP/zfp/build/zfp/lib -lh5zzfp -lhdf5 -lzfp -lm 
 
./test_error ................................................test_error: test_error.c:227: main: Assertion `check_hdf5_error_stack_for_string("chunk must have only 1...3 non-unity dimensions")' failed.
/bin/bash: line 12:  3392 Aborted                 (core dumped) $x 2>&1 > /dev/null
 [FAILED]
cd ../src; make  ZFP_HOME=/home/runner/work/H5Z-ZFP/H5Z-ZFP/zfp/build/zfp  PREFIX=/home/runner/work/H5Z-ZFP/H5Z-ZFP/install plugin
```

File: src/H5Zzfp.c (hunk 1 above)
Comment (brtnfld) [reply src/H5Zzfp.c:32]: Sorry this file should not change. let me fix the PR

**Comment by brtnfld** [comment]
I also tried it with zfp 0.5.0, but I could not get h5z-zfp to find the zfp build.

**Comment by markcmiller86** [comment]
> I also tried it with zfp 0.5.0, but I could not get h5z-zfp to find the zfp build with cmake.

You need to look for `inc` before looking for `include` to catch 0.5.0.

**Pull request closed without merging**